unity/Assets/Scripts/ContinuousMovement.cs

```diff
@@ -6,215 +6,207 @@
-            // Direction function for quaternion should just output target quaternion, since RotateTowards is used for addToProp
-            (target, current) => target,
-            // Distance Metric
-            (target, current) => Quaternion.Angle(current, target),
-            fixedDeltaTime,
-            returnToStartPropIfFailed
-        );
-    }
 
-    public static IEnumerator move(
-        PhysicsRemoteFPSAgentController controller,
-        CollisionListener collisionListener,
-        Transform moveTransform,
-        Vector3 targetPosition,
-        float fixedDeltaTime,
-        float unitsPerSecond,
-        bool returnToStartPropIfFailed = false,
-        bool localPosition = false
-    ) {
-        Func<Func<Transform, Vector3>, Action<Transform, Vector3>, Func<Transform, Vector3, Vector3>,IEnumerator> moveClosure = 
-            (get, set, next) => updateTransformPropertyFixedUpdate(
+        public static IEnumerator rotate(
+            PhysicsRemoteFPSAgentController controller,
+            CollisionListener collisionListener,
+            Transform moveTransform,
+            Quaternion targetRotation,
+            float fixedDeltaTime,
+            float radiansPerSecond,
+            bool returnToStartPropIfFailed = false
+        ) {
+            float degreesPerSecond = radiansPerSecond * 180.0f / Mathf.PI;
+            return updateTransformPropertyFixedUpdate(
                 controller,
                 collisionListener,
                 moveTransform,
-                targetPosition,
-                get,
-                set,
-                next,
-                (target, current) => (target - current).normalized,
-                (target, current) => Vector3.SqrMagnitude(target - current),
+                targetRotation,
+                // Get
+                (t) => t.rotation,
+                //  Set
+                (t, target) => t.rotation = target,
+                // Next
+                (t, target) => Quaternion.RotateTowards(t.rotation, target, fixedDeltaTime * degreesPerSecond),
+                // Direction function for quaternion should just output target quaternion, since RotateTowards is used for addToProp
+                (target, current) => target,
+                // Distance Metric
+                (target, current) => Quaternion.Angle(current, target),
                 fixedDeltaTime,
                 returnToStartPropIfFailed
-        );
-
-        if (localPosition) {
-            return moveClosure(
-                (t) => t.localPosition,
-                (t, pos) => t.localPosition = pos,
-                (t, direction) => t.localPosition + direction * unitsPerSecond * fixedDeltaTime
             );
         }
-        else {
 
-            return moveClosure(
-                (t) => t.position,
-                (t, pos) => t.position = pos,
-                (t, direction) => t.position + direction * unitsPerSecond * fixedDeltaTime
+        public static IEnumerator move(
```

Comment:
can we rename move to moveAgent and rotate to rotateAgent to distinguish them from moving and rotating the arm?

Reply:
These are all internal logic functions so we don’t necessarily need a name change for these. I’ll leave the final decision up to @mattdeitke

Reply:
Fixed (in the public API via. ArmAgentController.cs) in https://github.com/allenai/ai2thor/pull/697/commits/cb0241ccae8330002b38dc2ec0b40d234ce75c98.